File: back-end/app/Models/Cita.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

#[Fillable([
    'nombre_cliente',
    'telefono',
    'correo',
    'servicio_id',
    'promocion_id',
    'fecha_cita',
    'hora_cita',
    'notas_adicionales',
    'estado',
    'precio_final',
])]
class Cita extends Model
{
    protected $table = 'citas';

    protected function casts(): array
    {
        return [
            'fecha_cita' => 'date',
        ];
    }

    public function servicio(): BelongsTo
    {
        return $this->belongsTo(Servicio::class, 'servicio_id');
    }

    public function promocion(): BelongsTo
    {
        return $this->belongsTo(Promocion::class, 'promocion_id');
    }

    public function historial(): HasMany
    {
        return $this->hasMany(HistorialCita::class, 'cita_id');
    }
}

File: wearable/app/src/main/java/com/nailsstudio/wear/data/model/Cita.kt
package com.nailsstudio.wear.data.model

import com.google.gson.annotations.SerializedName

data class CrearCitaRequest(
    @SerializedName("nombre_cliente") val nombreCliente: String,
    val telefono: String? = null,
    val correo: String? = null,
    @SerializedName("servicio_id") val servicioId: Int? = null,
    @SerializedName("fecha_cita") val fechaCita: String,
    @SerializedName("hora_cita") val horaCita: String,
    @SerializedName("notas_adicionales") val notasAdicionales: String? = null,
)

data class CrearCitaResponse(
    val mensaje: String,
    val cita: Cita?,
)

data class Cita(
    val id: Long,
    @SerializedName("nombre_cliente") val nombreCliente: String,
    @SerializedName("fecha_cita") val fechaCita: String,
    @SerializedName("hora_cita") val horaCita: String,
    val estado: String,
)

data class ApiErrorResponse(
    val mensaje: String?,
    val errores: Map<String, List<String>>?,
)
